packages/quantic/force-app/main/default/lwc/quanticRecentQueriesList/__tests__/quanticRecentQueriesList.test.js

```diff
@@ -0,0 +1,447 @@
+        };
+      });
+
+      it('should display an empty list when #hideWhenEmpty is set to false', async () => {
+        const element = createTestComponent({
+          ...defaultOptions,
+          hideWhenEmpty: false,
+        });
+        updateRecentQueriesState();
+        await flushPromises();
+
+        const recentQueriesContainer = element.shadowRoot.querySelector(
+          selectors.quanticCardContainer
+        );
+        expect(recentQueriesContainer).not.toBeNull();
+        expect(recentQueriesContainer.title).toEqual(defaultRecentQueriesTitle);
+
+        const recentQueriesListItems = element.shadowRoot.querySelectorAll(
+          selectors.recentQueryItem
+        );
+        expect(recentQueriesListItems.length).toEqual(0);
+      });
+
+      it('should not display the recent queries card when #hideWhenEmpty is set to true', async () => {
+        const element = createTestComponent({
+          ...defaultOptions,
+          hideWhenEmpty: true,
+        });
+        await flushPromises();
+
+        const recentQueriesContainer = element.shadowRoot.querySelector(
+          selectors.quanticCardContainer
+        );
+        expect(recentQueriesContainer).toBeNull();
+      });
+    });
+
+    describe('when there are recent queries', () => {
+      const exampleQueries = ['query1', 'query2'];
+
+      it('should no longer display the placeholder component', async () => {
+        const element = createTestComponent();
+        recentQueriesListState = {
+          queries: exampleQueries,
+          maxLength: defaultMaxLength,
+        };
+        updateRecentQueriesState();
+        await flushPromises();
+
+        const placeholder = element.shadowRoot.querySelector(
+          selectors.placeholder
+        );
+
+        expect(placeholder).toBeNull();
+      });
+
+      it('should display the recent queries list with the default options', async () => {
+        recentQueriesListState = {
+          queries: exampleQueries,
+          maxLength: defaultMaxLength,
+        };
+        const element = createTestComponent();
+        updateRecentQueriesState();
+        await flushPromises();
+
+        const recentQueriesContainer = element.shadowRoot.querySelector(
+          selectors.quanticCardContainer
+        );
+        expect(recentQueriesContainer).not.toBeNull();
+        expect(recentQueriesContainer.title).toEqual(defaultRecentQueriesTitle);
+
+        const recentQueriesListItems = element.shadowRoot.querySelectorAll(
+          selectors.recentQueryItem
+        );
+        expect(recentQueriesListItems.length).toEqual(exampleQueries.length);
+
+        recentQueriesListItems.forEach((item, index) => {
+          const recentQueryText = item.querySelector(
+            selectors.recentQueryItemText
+          ).textContent;
+          expect(recentQueryText).toEqual(exampleQueries[index]);
+        });
+      });
+
+      describe('handling of the localStorage capabilities', () => {
```

Comment:
this describe is not really accurate, I think we can just remove it.`should use the correct initial state retrieved from the localstorage` and `should call #setItemInLocalStorage function with the proper parameters when the recent queries change in the state` are already well clarifying the test.